docs/rfcs/018-revisit-batches.md

```diff
@@ -0,0 +1,68 @@
+
+Currently a "Batch" is a loose and transient grouping of images. An image can belong to 1 batch at any given time, no history is maintained. This worked well for the initial use-case of bulk ingestion of images and tracking process for things like progress bars. However for more complex scenarios this connection is lacking and improving it 
+
+This RFC outlines proposal for improving batches, this will allow us to:
+* Get a list of assets in a batch, not just the current batch.
+* Extend named query support to filter on batchId(s)
+* Avoid issues with double counting completed or failed assets
+* Track which batches an images has been ingested in
+* Identify which assets failed within a batch
+* Reingest any failed assets in a batch (see [#402](https://github.com/dlcs/protagonist/issues/402))
+
+Note that not all of the above points are addressed here, some will need new API endpoints to be added building on the change outlines here.
+
+## Proposal
+
+The proposal is to introduce a new table, `BatchAssets`. This will track the assetIds per batch, maintaining the history of batch:asset. 
+
+The following is the proposed schema:
+
+| Column   | Type        | Description                                    |
+| -------- | ----------- | ---------------------------------------------- |
+| BatchId  | int         | Batch identifier                               |
+| AssetId  | text        | Asset identifier                               |
+| Status   | int         | Values of waiting/completed/error map to enum. |
+| Finished | timestamptz | Date when asset finished                       |
+
+> Note: 
+> * `Status` represents the status of processing asset in this batch, not _overall_ status of asset - it could have failed in batch 12345 but then been successful in batch 12999.
+> * `Finished` will store when the asset finished processing, we don't need to store `Created` as this will be stored against the batch.
+
+The `Batch` table will maintain it's current schema, this will serve as a quick status check for a batch.
+
+### Changes Required
+
+### API Endpoints
+
+The `/customers/{customerId}/queue/batches/{batchId}/images` endpoint will stay as-is and return all images currently associated with a batch (ie `select * from images where batch = @batch_id`). This will allow any existing integrations to continue working without changes. All other batch endpoints will stay as-is - we may update in the future but not now.
+
+A new endpoint, `/customers/{customerId}/queue/batches/{batchId}/assets`, will be introduced. This is effectively `v2` of the above endpoint but allows us to introduce the more appropriate `/assets` concept. This will use the new `BatchAssets` table and will allow us to return images from any batch, not just the current one, (ie `select i.* from images i inner join batch_assets ba on i.id = ba.asset_id where ba.batch_id = @batch_id`).

```

Comment:
I like this pattern. Do you think it would be worth doing a similar thing with spaces: `/spaces/{spaceId}/assets` with a `v2` schema - if built together they could solve some of the image-specific terminology, and also support some of the extra URLs created by different delivery channels (audio, video URLs)